### PR TITLE
Create the Activity Uploads page

### PR DESCRIPTION
**Pull request by hariniriyer**
Creating the activity upload page in developers.strava.com. This page has been created purely by copying over the contents in https://strava.github.io/api/v3/uploads/#post-file.
I have looked over to make sure all links work and all the text is accurate but would really appreciate it if someone could look for typos and other misleading information.

The landing page looks like the screenshot -

image

### CODE REVIEW

File: content/docs/uploads.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+          "external_id": "test.fit",
+          "error": null,
+          "status": "Your activity is still being processed.",
+          "activity_id": null
+        }
+
+##### Returns an Upload Status object
+This object will include an English language status. If success, it will indicate the data is still processing. If there was an error, it will describe the error, potentially containing HTML.
+
+Upon a successful submission the request will return 201 Created. If there was an error the request will return 400 Bad Request.
```
Comment (mindy-stivers) [inline content/docs/uploads.md]: Assuming I understand this correctly, can we be explicit and say the "201 Created" response does not mean the activity is fully uploaded and available, but only that it passed inititial checks and is now enqueued for processing?

As a side note, we should probably return 202 Accepted instead.

File: content/docs/uploads.md [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+
+
+## Check upload status
+
+Upon upload, Strava will respond with an upload ID. You may use this ID to poll the status of your upload. Strava recommends polling no more than once a second. Polling more frequently is unnecessary. The mean processing time is around 8 seconds.
```
Comment (mindy-stivers) [inline content/docs/uploads.md]: I think "upon upload" is a bit ambiguous, maybe we could say something along the lines of "An upload ID is included in the response to a successful upload POST request"?

The "Polling more frequently is unnecessary" sentence could be dropped entirely.

File: content/docs/uploads.md [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+          "id": 16486769,
+          "external_id": "testing.fit",
+          "error": null,
+          "status": "The created activity has been deleted.",
+          "activity_id": null
+        }
+
+
+## Errors
+One can determine if there was an error during upload by checking the error attribute for null. At this time error and status messages returned as part of the Upload object are human readable English. They many also include escaped HTML.
```
Comment (mindy-stivers) [inline content/docs/uploads.md]: Everywhere else in the page uses a familiar second person tone so the start of this line felt a little jarring.  Maybe starting with "You can determine..." would work?

Typo in "They **many** also include escaped HTML"  should probably read "They **may**..."

File: content/docs/uploads.md [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+
+All files are required to include a time with each trackpoint or record, as defined by the file format. Information such as lat/lng, elevation, heartrate, etc. is optional. Manual creation of activities without a data file is not currently supported by the API.
+
+If you feel your file is compatible with the standards but is still not uploading to Strava, please verify that it works with other fitness applications before contacting support.
+
+
+### FIT - Flexible and Interoperable Data Transfer
+Strava strives to comply with the FIT Activity File (FIT_FILE_TYPE = 4) spec as defined in the official [FIT SDK](https://www.thisisant.com/resources/fit).
+
+Many, many, attributes are defined by FIT. Below is an overview of those used by Strava.
```
Comment (mindy-stivers) [inline content/docs/uploads.md]: Don't need a comma between "many" and "attributes"

nitpick: "many, many" is a bit casual for written documentation... could just stick with "Below is an overview of FIT attributes used by Strava" or something along those lines.

**Comment by hariniriyer** [comment]
@mindy-stivers Great points. So this was a blind copy and paste from the other website in the description above. Feel free to go through that if that is easier.

File: content/docs/uploads.md (hunk 1 above)
Comment (hariniriyer) [reply content/docs/uploads.md]: I am not sure. @MateoAO might know more.

Comment (MateoAO) [reply content/docs/uploads.md]: I wonder if these two paragraphs could be collapsed? Something like.

This object will return a code and an English language status. On success, a 201 Created will be accompanied by a status of `success`. This indicates that the activity has been successfully accepted, but is still processing. On error, the 400 Bad Request will be accompanied by a status describing the error, potentially containing HTML.

Comment (MateoAO) [reply content/docs/uploads.md]: 202 would be proper, but would potentially break all sorts of apps.

**Comment by MateoAO** [comment]
I like the content updates!

File: content/docs/uploads.md (hunk 1 above)
Comment (hariniriyer) [reply content/docs/uploads.md]: I actually totally misread what Mindy said. Yeah, 202 is ideal. I will update the wording here.

File: content/docs/uploads.md (hunk 3 above)
Comment (hariniriyer) [reply content/docs/uploads.md]: Wow. Thanks!!

**Comment by hariniriyer** [comment]
@mindy-stivers PTAL

File: content/docs/uploads.md [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+## Device and elevation data
+For devices with a barometric altimeter the elevation data is taken as is from the file. Otherwise it is recomputed using the provided lat/lng points and an elevation database. [More information](https://strava.zendesk.com/entries/20965883-Elevation-for-Your-Activity).
+
+Device type is detected in all file types from standard ‘creator’ tags. These tags are then matched to a list of known devices. In some cases the name of the device will be displayed along with the activity details on [strava.com](http://strava.com/).
+
+A generic “with barometer” device is provided to force the system to use the elevation data from TCX and GPX file types. One only needs to add “with barometer” to the end of the creator name. For example, a TCX file would include something like:
+<pre>
+    <code>
+        `<Creator>
+            <Name>My Awesome App with barometer</Name>
```
Comment (MateoAO) [inline content/docs/uploads.md]: nit/favor: Change App to Device

File: content/docs/uploads.md [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+         </Creator>`
+    </code>
+</pre>
+
+and a GPX file should have an updated creator like:
+
+<pre>
+    <code> 
+        `< gpx version="1.1" creator="Best app ever with Barometer"/>`
```
Comment (MateoAO) [inline content/docs/uploads.md]: nit/favor: Change App to Device

File: layouts/index.html [hunk 7]
```diff
@@ -110,6 +110,15 @@ <h3 class="card-title">Rate Limiting</h3>
           </div>
         </div>
       </div>
+      <div class="col-md-4">
+        <div class="card">
+          <div class="card-block">
+            <h3 class="card-title">Uploads API</h3>
+            <p class="card-text">Understand the rules for usage.</p>
+            <a href="/docs/uploads" class="btn btn-primary">Learn</a>
```
Comment (MateoAO) [inline layouts/index.html]: You get the fun task of thinking of a CTA word that isn't already used. Learn is used elsewhere, iirc.

**Pull request merged**